### PR TITLE
Add optional secondary links to article download links

### PR DESCRIPTION
**Pull request by davidcmoulton**
Motivation: to allow a DAR download link to have a proximal explanatory link.

### CODE REVIEW

File: test/articledownloadlinkslist.html [hunk 1]
```diff
@@ -27,33 +27,71 @@
...
+        >Medlars</a></li>
+      </ul>
       <h3 class="article-download-links-list__heading">Open citations<span class="visuallyhidden"> (links to open the citations from this article in various online reference manager services)</span></h3>
-      <ul>
-        <li><a href="#" class="article-download-links-list__link">Mendeley</a></li>
-        <li><a href="#" class="article-download-links-list__link">ReadCube</a></li>
-        <li><a href="#" class="article-download-links-list__link">Papers App</a></li>
-        <li><a href="#" class="article-download-links-list__link">CiteUlike</a></li>
+      <ul class="article-download-list">
+        <li><a href="#" class="article-download-links-list__link"
```
Comment (giorgiosironi) [inline test/articledownloadlinkslist.html]: weird tags, are they generated? (or the expected value of a generation process?)

Comment (davidcmoulton) [reply test/articledownloadlinkslist.html]: Deliberately the same as the pattern library creates when it applies the data to the mustache template.

File: assets/sass/_normalize-overrides.scss [hunk 2]
```diff
@@ -61,7 +61,7 @@ p a:not(.additional-asset__link--download):not(.asset-viewer-inline__download_al
   @include discreet-link();
 }
 
-.article-section__body ul:not(.reference__abstracts) a {
+.article-section__body ul:not(.reference__abstracts) a:not(.article-download-links-list__secondary_link) {
```
Comment (thewilkybarkid) [inline assets/sass/_normalize-overrides.scss]: 😨

File: source/_patterns/01-molecules/components/article-download-links-list.yaml [hunk 3]
```diff
@@ -22,27 +22,56 @@ properties:
...
+              secondary:
+                type: object
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  url:
+                    type: string
+                    minLength: 1
+                  attributes:
```
Comment (thewilkybarkid) [inline source/_patterns/01-molecules/components/article-download-links-list.yaml]: Not used.

File: source/_patterns/01-molecules/components/article-download-links-list.yaml [hunk 4]
```diff
@@ -22,27 +22,56 @@ properties:
...
+                    items:
+                      type: object
+                      properties:
+                        key:
+                          type: string
+                          minLength: 1
+                        value:
+                          type: string
+                      required:
+                        - key
```
Comment (thewilkybarkid) [inline source/_patterns/01-molecules/components/article-download-links-list.yaml]: This isn't implemented.

File: assets/sass/_normalize-overrides.scss (hunk 2 above)
Comment (davidcmoulton) [reply assets/sass/_normalize-overrides.scss]: Yeah, the `discreet-link` handling isn't pretty.

**Pull request merged**